waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/compatibility/HiveCompatibleThriftHiveMetastoreIfaceFactory.java

```diff
@@ -71,7 +71,7 @@ private Object invokeCompatibility(Method method, Object[] args) throws Throwabl
 
     private Class<?>[] getTypes(Object[] args) {
       if (args == null) {
-        return (Class<?>[]) args;
+        return new Class<?>[] {};
```

Comment:
`args` is already null, so this is equivalent, but doesn't load a null variable.

Reply:
Yeah, this is clearer.